src/views/Pools/components/CakeVaultCard/VaultStakeModal.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { Modal, Text, Flex, Image, Button, Slider, BalanceInput, AutoRenewIcon } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+import { BASE_EXCHANGE_URL } from 'config'
+import { useCakeVaultContract } from 'hooks/useContract'
+import useTheme from 'hooks/useTheme'
+import useWithdrawalFeeTimer from 'hooks/cakeVault/useWithdrawalFeeTimer'
+import { VaultFees } from 'hooks/cakeVault/useGetVaultFees'
+import BigNumber from 'bignumber.js'
+import { getFullDisplayBalance, formatNumber, getDecimalAmount } from 'utils/formatBalance'
+import useToast from 'hooks/useToast'
+import { Pool } from 'state/types'
+import { VaultUser } from 'views/Pools/types'
+import { convertCakeToShares } from '../../helpers'
+import FeeSummary from './FeeSummary'
+
+interface VaultStakeModalProps {
+  pool: Pool
+  stakingMax: BigNumber
+  stakingTokenPrice: number
+  account: string
+  userInfo: VaultUser
+  isRemovingStake?: boolean
+  pricePerFullShare?: BigNumber
+  vaultFees?: VaultFees
+  setLastUpdated: () => void
+  onDismiss?: () => void
+}
+
+const StyledButton = styled(Button)`
+  flex-grow: 1;
+`
+
+const VaultStakeModal: React.FC<VaultStakeModalProps> = ({
+  pool,
+  stakingMax,
+  stakingTokenPrice,
+  pricePerFullShare,
+  account,
+  userInfo,
+  isRemovingStake = false,
+  vaultFees,
+  onDismiss,
+  setLastUpdated,
+}) => {
+  const { stakingToken } = pool
+  const cakeVaultContract = useCakeVaultContract()
+  const TranslateString = useI18n()
+  const { theme } = useTheme()
+  const { toastSuccess, toastError } = useToast()
+  const [pendingTx, setPendingTx] = useState(false)
+  const [stakeAmount, setStakeAmount] = useState('')
+  const [percent, setPercent] = useState(0)
+  const { hasPerformanceFee } = useWithdrawalFeeTimer(parseInt(userInfo.lastDepositedTime))
+  const usdValueStaked = stakeAmount && formatNumber(parseFloat(stakeAmount) * stakingTokenPrice)
```

Comment:
arithmetic operation on balances must use Bignumber

Reply:
Cheers Doge, going to go through and check the vault against the kind of changes I made following your reviews in the pool tomorrow (once shipped). I'll be able to strip out some repeated code, too. https://github.com/pancakeswap/pancake-frontend/commit/dfa6bc5aa092e1b9cbd1775d013df125568f660a